src/odb.c

```diff
@@ -1521,6 +1521,25 @@ void git_odb_backend_data_free(git_odb_backend *backend, void *data)
 	git__free(data);
 }
 
+int git_odb_set_backend_priority(git_odb *db, size_t index, int priority)
+{
+	size_t i;
+	assert(db);
+
+	if (index > db->backends.length - 1) {
+		git_error_set(GIT_ERROR_INVALID, "invalid odb backend index");
+		return -1;
+	}
+
+	for (i = 0; i < db->backends.length; ++i){
```

Comment:
Why do you have this loop?  You should be able to `get_vector_get` using the specified `index`.

Reply:
Once we sort the vector at the end of this function, we can assume the backends will always be sorted based on priority. This will allow us to get the entry at a given index.

Reply:
Fixed this in a new commit with a call to `git_vector_get`.